src/controller/index.js

```diff
@@ -0,0 +1,60 @@
+import BaseballModel from '../model/index.js';
+import { InputView, OutputView } from '../view/index.js';
+
+class BaseBallController {
+  /** * @type {string} */
+  static RETRY = '1';
+
+  /** * @type {number} */
+  static END_POINT = 3;
+
+  constructor() {
+    OutputView.printStart();
+  }
+
+  /**
+   * @async
+   * @returns {Promise<void>}
+   */
+  async run() {
+    const gameNumbers = BaseballModel.generateGameNumbers();
+    await BaseBallController.#guessNumber(gameNumbers);
+    const userAnswer = await InputView.readGameCommand();
+
+    if (userAnswer === BaseBallController.RETRY) {
+      this.run();
+    }
+  }
```

Comment:
```suggestion
  async run() {
    const computerNumbers = BaseballModel.generateComputerNumbers();
    while (true) {
      const userNumbers = await InputView.readGameNumbers();
      const score = BaseballModel.compareUserWithComputerNumbers(userNumbers, computerNumbers);

      OutputView.printHint(score);
      if (score.strike === this.END_POINT) {
        OutputView.printSuccess();
        break;
      }
    }

    const userAnswer = await InputView.readGameCommand();
    
    if (userAnswer === BaseBallController.RETRY) {
      this.run();
    }
  }
```

이렇게 구현하는게 더 좋은 것 같습니다.

Reply:
셀프 피드백 좋습니다..